### PR TITLE
fix: can not upgrade karmada controller manager deployment with helm upgrade

### LINKED ISSUE #3813: can't upgrade karmada controller manager deployment with Helm upgrade
**What happened**:
override controllerManager.extraCommandArgs not effect on Helm upgrade 

**What you expected to happen**:
I can override controllerManager.extraCommandArgs

**How to reproduce it (as minimally and precisely as possible)**:
1. helm install
2. helm upgrade -n karmada-system karmada . -f value-override.yml 
```
controllerManager:
  image:
    tag: v1.6.1
  extraCommandArgs:
    skipped-propagating-namespaces: namespace1,namespace2
```

**Anything else we need to know?**:
The root case is helm post-install hook only apply deployment in install
https://github.com/karmada-io/karmada/blob/d27faeac953043c2274bb8787bbbaaafa7d0dcd6/charts/karmada/templates/karmada-controller-manager.yaml#L12

**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

### PR DESCRIPTION
**Pull request by CoderTH**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

in this issue,Error occurs when using helm upgrade karmada, here add post-upgrade in karmada-controller-manager.yaml

**Which issue(s) this PR fixes**:
Fixes #3813 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CoderTH** [comment]
image
image
image
test successful

File: charts/karmada/templates/karmada-controller-manager.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ metadata:
   labels:
   {{- include "karmada.cm.labels" . | nindent 4 }}
   annotations:
-    "helm.sh/hook": post-install
+    "helm.sh/hook": post-install,post-upgrade
```
Comment (Poor12) [inline charts/karmada/templates/karmada-controller-manager.yaml:12]: Ref to #3813, I think deleting may be better.

Comment (CoderTH) [reply charts/karmada/templates/karmada-controller-manager.yaml:12]: Ok, I'll delete it

**Comment by CoderTH** [comment]
done, please review @Poor12 @calvin0327

**Comment by calvin0327** [comment]
I agree with @Poor12 , if  rollback version problem cannot be resolved, we would rather remove this hook.

I can stand the karmada-controller-manager restart a few times.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**